pkg/pipeline/pipeline_project_integration.go

```diff
@@ -0,0 +1,205 @@
+		Type:         schema.TypeInt,
+		Required:     true,
+		ValidateFunc: validation.IntAtLeast(0),
+		Description:  "The Id of the master integration.",
+	},
+	"master_integration_name": {
+		Type:         schema.TypeString,
+		Optional:     true,
+		ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotEmpty),
+		Description:  "The name of the master integration.",
+	},
+}
+
+func unpackProject(data *util.ResourceData, key string) ProjectJSON {
+	var project ProjectJSON
+	input := data.Get(key).(map[string]interface{})
+	project.Key = input["key"].(string)
+	project.Name = input["name"].(string)
+
+	return project
+}
+
+func packProject(d *schema.ResourceData, schemaKey string, project ProjectJSON) []error {
+	var errors []error
+	if (ProjectJSON{}) == project {
+		return errors
+	}
+	setValue := util.MkLens(d)
+	errors = append(errors, setValue(schemaKey, project)...)
+	return errors
+}
+
+func unpackProjectIntegration(data *schema.ResourceData, formJsonValues []FormJSONValues) (ProjectIntegration, error) {
+	d := &util.ResourceData{data}
+
+	projectIntegration := ProjectIntegration{
+		Name:                  d.GetString("name", false),
+		ProjectId:             d.GetInt("project_id", false),
+		MasterIntegrationId:   d.GetInt("master_integration_id", false),
+		MasterIntegrationName: d.GetString("master_integration_name", false),
+		Environments:          d.GetSet("environments"),
+		IsInternal:            d.GetBool("is_internal", false),
+		Project:               unpackProject(d, "project"),
+		FormJSONValues:        formJsonValues,
+	}
+	return projectIntegration, nil
+}
+
+func packProjectIntegration(d *schema.ResourceData, projectIntegration ProjectIntegration) []error {
+	setValue := util.MkLens(d)
+
+	setValue("name", projectIntegration.Name)
+	setValue("master_integration_id", projectIntegration.MasterIntegrationId)
+	setValue("master_integration_name", projectIntegration.MasterIntegrationName)
+	setValue("environments", projectIntegration.Environments)
+	errors := setValue("is_internal", projectIntegration.IsInternal)
+	errors = append(errors, packProject(d, "project", projectIntegration.Project)...)
+
+	return errors
+}
+
+func setUniqueIntegrationNameAndId(d *schema.ResourceData, name string, id int) []error {
+	setValue := util.MkLens(d)
+	setValue("master_integration_name", name)
+	errors := setValue("master_integration_id", id)
+	return errors
+}
+
+func readProjectIntegration(data *schema.ResourceData, m interface{}) ([]FormJSONValues, error) {
+	projectIntegration := ProjectIntegration{}
+	_, err := m.(*resty.Client).R().
+		SetResult(&projectIntegration).
+		Get(projectIntegrationsUrl + "/" + data.Id())
+	if err != nil {
+		return nil, err
+	}
+	errors := packProjectIntegration(data, projectIntegration)
+	if len(errors) > 0 {
+		return nil, errors[0]
+	}
+	return projectIntegration.FormJSONValues, nil
+}
+
+func createProjectIntegration(data *schema.ResourceData, m interface{}, formValues []FormJSONValues) error {
```

Comment:
You have to use `XXXContext` functions - these are deprecated. Not difficult to do